core/src/main/java/org/elasticsearch/rest/action/cat/AbstractCatAction.java

```diff
@@ -54,6 +54,8 @@ public AbstractCatAction(Settings settings) {
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
         boolean helpWanted = request.paramAsBoolean("help", false);
         if (helpWanted) {
+            // consume all request parameters for 'help' to prevent an IllegalArgumentException
+            request.params().keySet().forEach(request::param);
```

Comment:
Shouldn't this only be consuming the `help` key? Any random other params should still cause an error.